server/app/views/applicant/ApplicantLayout.java

```diff
@@ -305,7 +301,7 @@ private DivTag authDisplaySection(String userName, Messages messages) {
                   .withHref(createAnAccountLink)
                   .withClasses(ApplicantStyles.LINK)));
     } else {
-      String loggedInAsMessage = messages.at(MessageKey.USER_NAME.getKeyName(), userName);
+      String loggedInAsMessage = messages.at(MessageKey.USER_NAME.getKeyName(), userName.get());
```

Comment:
Doesn't look like you're checking if the Optional isPresent before dereferencing?

Reply:
It was guaranteed to be present is `isGuest` returns false. But I realized this is bad coupling; `isGuest` was essentially just checking is the Optional is empty in a thinly veiled way.

I replaced that function with calls to `isEmpty()` directly, and a reference to #4626 which will address it.